src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -429,13 +429,35 @@ private async Task<Expression> RecurseChainedExpression(ChainedExpression expres
 
             try
             {
-                (IReadOnlyList<T> results, string nextContinuationToken) = await _fhirDataStore.ExecuteDocumentQueryAsync<T>(sqlQuerySpec, feedOptions, continuationToken, searchOptions.MaxItemCountSpecifiedByClient, feedOptions.MaxConcurrency == null ? searchEnumerationTimeoutOverrideIfSequential : null, cancellationToken);
-
-                if (queryPartitionStatistics != null && messagesList != null)
+                (IReadOnlyList<T> results, string nextContinuationToken) = (null, null);
+                var desiredItemCount = feedOptions.MaxItemCount * CosmosFhirDataStore.ExecuteDocumentQueryAsyncMinimumFillFactor;
+
+                // Set timeout for sequential query execution
+                if (feedOptions.MaxConcurrency == null &&
+                    _cosmosConfig.ParallelQueryOptions.EnableConcurrencyIfQueryExceedsTimeLimit == true &&
+                    string.IsNullOrEmpty(searchOptions.ContinuationToken) &&
+                    searchEnumerationTimeoutOverrideIfSequential == null)
```

Comment:
While testing I also realized that `continuationToken` is being passed into this function, this option is also only available when this is null.

```suggestion
                if (feedOptions.MaxConcurrency == null &&
                    _cosmosConfig.ParallelQueryOptions.EnableConcurrencyIfQueryExceedsTimeLimit == true &&
                    string.IsNullOrEmpty(searchOptions.ContinuationToken) &&
                    string.IsNullOrEmpty(continuationToken) &&
                    searchEnumerationTimeoutOverrideIfSequential == null)
```

Reply:
Thanks! Done